### PR TITLE
Set read size to 1023 when read files for nuttx compatibility

### PR DESCRIPTION
**Pull request by chokobole**
IoT.js-DCO-1.0-Signed-off-by: wonyong.kim wonyong.kim@samsung.com

### CODE REVIEW

File: src/js/fs.js [hunk 1]
```diff
@@ -221,8 +221,8 @@ fs.readFile = function(path, callback) {
 
   var read = function() {
     // Read segment of data.
-    var buffer = new Buffer(1024);
-    fs.read(fd, buffer, 0, 1024, -1, afterRead);
+    var buffer = new Buffer(1023);
+    fs.read(fd, buffer, 0, 1023, -1, afterRead);
```
Comment (yichoi) [inline src/js/fs.js]: I don't think this value is related with above definition

File: src/iotjs_def.h [hunk 2]
```diff
@@ -20,7 +20,7 @@
 
 #ifndef IOTJS_MAX_READ_BUFFER_SIZE
  #ifdef __NUTTX__
-  #define IOTJS_MAX_READ_BUFFER_SIZE 1024
+  #define IOTJS_MAX_READ_BUFFER_SIZE 1023
```
Comment (yichoi) [inline src/iotjs_def.h]: Please add reference why 1023 is valid

**Comment by LaszloLango** [comment]
Is it possible to write a test for this issue?

**Comment by chokobole** [comment]
@LaszloLango I raised an issue about this..

**Pull request merged**